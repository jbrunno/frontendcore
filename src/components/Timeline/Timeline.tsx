import React, { Ref } from 'react';
import {
  Timeline as MuiTimeline,
  TimelineProps as MuiTimelineProps,
} from '@mui/lab';

export type TimelineProps = MuiTimelineProps & {
  ref?: Ref<HTMLUListElement>;
};

export function Timeline(props: TimelineProps) {
  return <MuiTimeline {...props} />;
}

export default Timeline;
