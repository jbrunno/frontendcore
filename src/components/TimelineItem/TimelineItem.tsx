import React from 'react';
import {
  TimelineItem as MuiTimelineItem,
  TimelineItemProps as MuiTimelineItemProps,
} from '@mui/lab';

export type TimelineItemProps = MuiTimelineItemProps;

export function TimelineItem(props: TimelineItemProps) {
  return <MuiTimelineItem {...props} />;
}

export default TimelineItem;
