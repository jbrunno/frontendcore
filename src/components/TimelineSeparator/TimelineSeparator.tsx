import React from 'react';
import {
  TimelineSeparator as MuiTimelineSeparator,
  TimelineSeparatorProps as MuiTimelineSeparatorProps,
} from '@mui/lab';

export type TimelineSeparatorProps = MuiTimelineSeparatorProps;

export function TimelineSeparator(props: TimelineSeparatorProps) {
  return <MuiTimelineSeparator {...props} />;
}

export default TimelineSeparator;
