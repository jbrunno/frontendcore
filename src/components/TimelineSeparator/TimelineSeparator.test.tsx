import React from 'react';
import { render } from '@test-utils/component-testing';
import Timeline from '../Timeline/Timeline';
import TimelineItem from '../TimelineItem/TimelineItem';
import TimelineSeparator from './TimelineSeparator';

test('renders the TimelineConnector', () => {
  const { container } = render(
    <Timeline position="left">
      <TimelineItem>
        <TimelineSeparator />
      </TimelineItem>
    </Timeline>,
  );
  expect(container).toMatchSnapshot();
});
