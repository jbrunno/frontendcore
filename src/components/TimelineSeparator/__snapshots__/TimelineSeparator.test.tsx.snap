// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the TimelineConnector 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px 16px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-1 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  min-height: 70px;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.emotion-1:before {
  content: "";
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 6px 16px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <ul
    class="MuiTimeline-root MuiTimeline-positionLeft emotion-0"
  >
    <li
      class="MuiTimelineItem-root MuiTimelineItem-positionLeft MuiTimelineItem-missingOppositeContent emotion-1"
    >
      <div
        class="MuiTimelineSeparator-root emotion-2"
      />
    </li>
  </ul>
</div>
`;
